File: examples/multi-platform/previews/ios/Sources/DesignTokensPreview/TokenShowcase.swift
// Token Showcase — SwiftUI view showing all Spectrum design tokens
// Uses SpectrumTokens from the Dispersa-generated Swift file

import SwiftUI

struct TokenShowcase: View {
    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(alignment: .leading, spacing: 32) {
                    semanticColorsSection
                    spacingSection
                    shadowSection
                }
                .padding(24)
            }
            .background(SpectrumTokens.Colors.backgroundBase)
            .navigationTitle("Spectrum Tokens")
        }
    }

    // MARK: - Semantic Colors

    private var semanticColorsSection: some View {
        VStack(alignment: .leading, spacing: 16) {
            Text("Semantic Colors")
                .font(.title2.weight(.bold))
                .foregroundColor(SpectrumTokens.Colors.textPrimary)

            LazyVGrid(columns: [GridItem(.adaptive(minimum: 140), spacing: 12)], spacing: 12) {
                colorCard("Accent", color: SpectrumTokens.Colors.accentBackgroundDefault)
                colorCard("Accent Hover", color: SpectrumTokens.Colors.accentBackgroundHover)
                colorCard("Negative", color: SpectrumTokens.Colors.negativeBackgroundDefault)
                colorCard("Positive", color: SpectrumTokens.Colors.positiveBackgroundDefault)
                colorCard("Notice", color: SpectrumTokens.Colors.noticeBackgroundDefault)
                colorCard("Neutral", color: SpectrumTokens.Colors.neutralBackgroundDefault)
                colorCard("Disabled", color: SpectrumTokens.Colors.disabledBackground)
                colorCard("Focus", color: SpectrumTokens.Colors.focusIndicator)
            }

            Text("Text Colors")
                .font(.headline)
                .foregroundColor(SpectrumTokens.Colors.textPrimary)
                .padding(.top, 8)

            VStack(alignment: .leading, spacing: 8) {
                Text("Primary text")
                    .foregroundColor(SpectrumTokens.Colors.textPrimary)
                Text("Secondary text")
                    .foregroundColor(SpectrumTokens.Colors.textSecondary)
                Text("Disabled text")
                    .foregroundColor(SpectrumTokens.Colors.textDisabled)
                HStack {
                    Text("Text on accent")
                        .foregroundColor(SpectrumTokens.Colors.textOnAccent)
                        .padding(.horizontal, 12)
                        .padding(.vertical, 6)
                        .background(SpectrumTokens.Colors.accentBackgroundDefault)
                        .clipShape(RoundedRectangle(cornerRadius: 6))
                }
            }

            Text("Backgrounds")
                .font(.headline)
                .foregroundColor(SpectrumTokens.Colors.textPrimary)
                .padding(.top, 8)

            LazyVGrid(columns: [GridItem(.adaptive(minimum: 140), spacing: 12)], spacing: 12) {
                backgroundCard("Base", color: SpectrumTokens.Colors.backgroundBase)
                backgroundCard("Layer 1", color: SpectrumTokens.Colors.backgroundLayer1)
                backgroundCard("Layer 2", color: SpectrumTokens.Colors.backgroundLayer2)
            }
        }
    }

    // MARK: - Spacing

    private var spacingSection: some View {
        VStack(alignment: .leading, spacing: 16) {
            Text("Spacing Scale")
                .font(.title2.weight(.bold))
                .foregroundColor(SpectrumTokens.Colors.textPrimary)

            VStack(alignment: .leading, spacing: 8) {
                spacingBar("50", width: SpectrumTokens.Spacing._50)
                spacingBar("75", width: SpectrumTokens.Spacing._75)
                spacingBar("100", width: SpectrumTokens.Spacing._100)
                spacingBar("200", width: SpectrumTokens.Spacing._200)
                spacingBar("300", width: SpectrumTokens.Spacing._300)
                spacingBar("400", width: SpectrumTokens.Spacing._400)
                spacingBar("500", width: SpectrumTokens.Spacing._500)
                spacingBar("600", width: SpectrumTokens.Spacing._600)
            }
        }
    }

    // MARK: - Shadows

    private var shadowSection: some View {
        VStack(alignment: .leading, spacing: 16) {
            Text("Shadows")
                .font(.title2.weight(.bold))
                .foregroundColor(SpectrumTokens.Colors.textPrimary)

            LazyVGrid(columns: [GridItem(.adaptive(minimum: 140), spacing: 24)], spacing: 24) {
                shadowCard("100", radius: 4, y: 1, opacity: 0.12)
                shadowCard("200", radius: 16, y: 4, opacity: 0.16)
                shadowCard("300", radius: 28, y: 8, opacity: 0.24)
            }
        }
    }

    // MARK: - Helper Views

    private func colorCard(_ label: String, color: Color) -> some View {
        VStack(spacing: 8) {
            RoundedRectangle(cornerRadius: 8)
                .fill(color)
                .frame(height: 60)
            Text(label)
                .font(.caption)
                .foregroundColor(SpectrumTokens.Colors.textSecondary)
        }
    }

    private func backgroundCard(_ label: String, color: Color) -> some View {
        VStack(spacing: 8) {
            RoundedRectangle(cornerRadius: 8)
                .fill(color)
                .frame(height: 60)
                .overlay(
                    RoundedRectangle(cornerRadius: 8)
                        .stroke(SpectrumTokens.Colors.borderDefault, lineWidth: 1)
                )
            Text(label)
                .font(.caption)
                .foregroundColor(SpectrumTokens.Colors.textSecondary)
        }
    }

    private func spacingBar(_ label: String, width: CGFloat) -> some View {
        HStack(spacing: 12) {
            Text(label)
                .font(.caption.monospacedDigit())
                .foregroundColor(SpectrumTokens.Colors.textSecondary)
                .frame(width: 30, alignment: .trailing)
            RoundedRectangle(cornerRadius: 3)
                .fill(SpectrumTokens.Colors.accentBackgroundDefault)
                .frame(width: width * 3, height: 16)
            Text("\(Int(width))px")
                .font(.caption.monospacedDigit())
                .foregroundColor(SpectrumTokens.Colors.textDisabled)
        }
    }

    private func shadowCard(_ label: String, radius: CGFloat, y: CGFloat, opacity: Double) -> some View {
        VStack(spacing: 8) {
            RoundedRectangle(cornerRadius: 8)
                .fill(SpectrumTokens.Colors.backgroundBase)
                .frame(height: 80)
                .shadow(color: .black.opacity(opacity), radius: radius, x: 0, y: y)
            Text("shadow-\(label)")
                .font(.caption)
                .foregroundColor(SpectrumTokens.Colors.textSecondary)
        }
    }
}

#Preview {
    TokenShowcase()
}

File: examples/multi-platform/previews/android/src/main/kotlin/com/example/spectrum/preview/TokenShowcase.kt
package com.example.spectrum.preview

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.spectrum.tokens.SpectrumTokens

@Composable
fun TokenShowcase() {
    LazyColumn(
        modifier = Modifier.padding(24.dp),
        verticalArrangement = Arrangement.spacedBy(32.dp),
    ) {
        item { SemanticColorsSection() }
        item { SpacingSection() }
        item { ShadowSection() }
    }
}

// ── Semantic Colors ──────────────────────────────────────────────

@Composable
private fun SemanticColorsSection() {
    Column(verticalArrangement = Arrangement.spacedBy(16.dp)) {
        SectionTitle("Semantic Colors")

        LazyVerticalGrid(
            columns = GridCells.Adaptive(minSize = 140.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp),
            modifier = Modifier.height(280.dp),
        ) {
            item { ColorCard("Accent", SpectrumTokens.Colors.accentBackgroundDefault) }
            item { ColorCard("Accent Hover", SpectrumTokens.Colors.accentBackgroundHover) }
            item { ColorCard("Negative", SpectrumTokens.Colors.negativeBackgroundDefault) }
            item { ColorCard("Positive", SpectrumTokens.Colors.positiveBackgroundDefault) }
            item { ColorCard("Notice", SpectrumTokens.Colors.noticeBackgroundDefault) }
            item { ColorCard("Neutral", SpectrumTokens.Colors.neutralBackgroundDefault) }
            item { ColorCard("Disabled", SpectrumTokens.Colors.disabledBackground) }
            item { ColorCard("Focus", SpectrumTokens.Colors.focusIndicator) }
        }

        Text(
            text = "Text Colors",
            fontSize = 16.sp,
            fontWeight = FontWeight.Bold,
            color = SpectrumTokens.Colors.textPrimary,
        )

        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            Text("Primary text", color = SpectrumTokens.Colors.textPrimary)
            Text("Secondary text", color = SpectrumTokens.Colors.textSecondary)
            Text("Disabled text", color = SpectrumTokens.Colors.textDisabled)
            Text(
                text = "Text on accent",
                color = SpectrumTokens.Colors.textOnAccent,
                modifier = Modifier
                    .background(
                        SpectrumTokens.Colors.accentBackgroundDefault,
                        RoundedCornerShape(6.dp),
                    )
                    .padding(horizontal = 12.dp, vertical = 6.dp),
            )
        }

        Text(
            text = "Backgrounds",
            fontSize = 16.sp,
            fontWeight = FontWeight.Bold,
            color = SpectrumTokens.Colors.textPrimary,
        )

        LazyVerticalGrid(
            columns = GridCells.Adaptive(minSize = 140.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp),
            modifier = Modifier.height(100.dp),
        ) {
            item { BackgroundCard("Base", SpectrumTokens.Colors.backgroundBase) }
            item { BackgroundCard("Layer 1", SpectrumTokens.Colors.backgroundLayer1) }
            item { BackgroundCard("Layer 2", SpectrumTokens.Colors.backgroundLayer2) }
        }
    }
}

// ── Spacing ──────────────────────────────────────────────────────

@Composable
private fun SpacingSection() {
    Column(verticalArrangement = Arrangement.spacedBy(16.dp)) {
        SectionTitle("Spacing Scale")

        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            SpacingBar("50", SpectrumTokens.Spacing._50)
            SpacingBar("75", SpectrumTokens.Spacing._75)
            SpacingBar("100", SpectrumTokens.Spacing._100)
            SpacingBar("200", SpectrumTokens.Spacing._200)
            SpacingBar("300", SpectrumTokens.Spacing._300)
            SpacingBar("400", SpectrumTokens.Spacing._400)
            SpacingBar("500", SpectrumTokens.Spacing._500)
            SpacingBar("600", SpectrumTokens.Spacing._600)
        }
    }
}

// ── Shadows ──────────────────────────────────────────────────────

@Composable
private fun ShadowSection() {
    Column(verticalArrangement = Arrangement.spacedBy(16.dp)) {
        SectionTitle("Shadows")

        LazyVerticalGrid(
            columns = GridCells.Adaptive(minSize = 140.dp),
            horizontalArrangement = Arrangement.spacedBy(24.dp),
            verticalArrangement = Arrangement.spacedBy(24.dp),
            modifier = Modifier.height(130.dp),
        ) {
            item { ShadowCard("100", elevation = 2.dp) }
            item { ShadowCard("200", elevation = 8.dp) }
            item { ShadowCard("300", elevation = 16.dp) }
        }
    }
}

// ── Helpers ──────────────────────────────────────────────────────

@Composable
private fun SectionTitle(text: String) {
    Text(
        text = text,
        fontSize = 20.sp,
        fontWeight = FontWeight.Bold,
        color = SpectrumTokens.Colors.textPrimary,
    )
}

@Composable
private fun ColorCard(label: String, color: androidx.compose.ui.graphics.Color) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(60.dp)
                .clip(RoundedCornerShape(8.dp))
                .background(color),
        )
        Text(label, fontSize = 12.sp, color = SpectrumTokens.Colors.textSecondary)
    }
}

@Composable
private fun BackgroundCard(label: String, color: androidx.compose.ui.graphics.Color) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(60.dp)
                .clip(RoundedCornerShape(8.dp))
                .background(color)
                .border(1.dp, SpectrumTokens.Colors.borderDefault, RoundedCornerShape(8.dp)),
        )
        Text(label, fontSize = 12.sp, color = SpectrumTokens.Colors.textSecondary)
    }
}

@Composable
private fun SpacingBar(label: String, size: androidx.compose.ui.unit.Dp) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Text(
            text = label,
            fontSize = 12.sp,
            color = SpectrumTokens.Colors.textSecondary,
            modifier = Modifier.width(30.dp),
        )
        Box(
            modifier = Modifier
                .width(size * 3)
                .height(16.dp)
                .clip(RoundedCornerShape(3.dp))
                .background(SpectrumTokens.Colors.accentBackgroundDefault),
        )
        Text(
            text = "${size.value.toInt()}px",
            fontSize = 12.sp,
            color = SpectrumTokens.Colors.textDisabled,
        )
    }
}

@Composable
private fun ShadowCard(label: String, elevation: androidx.compose.ui.unit.Dp) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .height(80.dp)
                .shadow(elevation, RoundedCornerShape(8.dp)),
            shape = RoundedCornerShape(8.dp),
            colors = CardDefaults.cardColors(
                containerColor = SpectrumTokens.Colors.backgroundBase,
            ),
        ) {}
        Text("shadow-$label", fontSize = 12.sp, color = SpectrumTokens.Colors.textSecondary)
    }
}

@Preview(showBackground = true)
@Composable
private fun TokenShowcasePreview() {
    TokenShowcase()
}
